### PR TITLE
VideoBackends:Vulkan: Prevent freezes during window resize on Linux

### PR DESCRIPTION
**Pull request by TellowKrinkle**
If `vkAcquireNextImageKHR` failed, it wouldn't signal the semaphore it was given, but we'd mark it as used anyways, making us wait for it forever.

Also handles the case where `vkAcquireNextImageKHR` returns `VK_ERROR_OUT_OF_DATE_KHR` immediately after resize (e.g. because the user is still resizing the window) by skipping presentation and trying again next frame.

Thanks to @CasualPokePlayer for tracking down the root cause for both of these issues

### CODE REVIEW

**Comment by dreamsyntax** [comment]
@dolphin-emu-bot rebuild freebsd

**Comment by mitaclaw** [comment]
Tested on Manjaro Linux using an NVIDIA Corporation, NVIDIA GeForce GTX 980 Ti/PCIe/SSE2, 4.6.0 NVIDIA 550.107.02 (using proprietary drivers).

Finally, a working fix! Resizing the render widget with and without the render to main window option no longer deadlocks on my machine. Both debug and release builds work great.

File: Source/Core/VideoCommon/Present.cpp [hunk 1]
```diff
@@ -844,10 +844,10 @@ void Presenter::Present()
   UpdateDrawRectangle();
 
   g_gfx->BeginUtilityDrawing();
-  g_gfx->BindBackbuffer({{0.0f, 0.0f, 0.0f, 1.0f}});
+  bool ok = g_gfx->BindBackbuffer({{0.0f, 0.0f, 0.0f, 1.0f}});
```
Comment (dreamsyntax) [inline Source/Core/VideoCommon/Present.cpp:847]: Minor nitpick: why is this 'ok'? I don't know what else you'd call it here though.

**Review by dreamsyntax — APPROVED** [review]
Tested on Arch
NVIDIA RTX 4080M | 560.35.03

Works great! No more deadlock/crash.

Resizing while detached from the main window is a bit funky, but that's unrelated to the crash.
Great work!

File: Source/Core/VideoCommon/Present.cpp (hunk 1 above)
Comment (TellowKrinkle) [reply Source/Core/VideoCommon/Present.cpp:847]: Because I also didn't know what else to call it

Comment (iwubcode) [reply Source/Core/VideoCommon/Present.cpp:847]: `backbuffer_bound`

Comment (iwubcode) [reply Source/Core/VideoCommon/Present.cpp:847]:
```suggestion
  const bool backbuffer_bound = g_gfx->BindBackbuffer({{0.0f, 0.0f, 0.0f, 1.0f}});
```

File: Source/Core/VideoBackends/Vulkan/VKGfx.cpp [hunk 2]
```diff
@@ -284,10 +284,18 @@ void VKGfx::BindBackbuffer(const ClearColor& clear_color)
 
     res = m_swap_chain->AcquireNextImage();
     if (res != VK_SUCCESS && res != VK_SUBOPTIMAL_KHR)
-      PanicAlertFmt("Failed to grab image from swap chain: {:#010X} {}", Common::ToUnderlying(res),
-                    VkResultToString(res));
+    {
+      if (res == VK_ERROR_OUT_OF_DATE_KHR)
+        INFO_LOG_FMT(VIDEO, "Swapchain still out of date, will try again next frame...");
+      else
```
Comment (iwubcode) [inline Source/Core/VideoBackends/Vulkan/VKGfx.cpp:290]: Multilines should use curly braces

**Review by iwubcode — APPROVED** [review]
Minor nitpicks but code LGTM.  Tested on Windows, no regressions.

**Comment by mitaclaw** [comment]
Once merged, this resolves https://bugs.dolphin-emu.org/issues/13523.

**Comment by homeisfar** [comment]
I ran this through my test that I came up with to test my own implementation. Your implementation works great. Great work, TellowKrinkle

**Comment by JosJuice** [comment]
Let's wait for it to be fixed.

**Comment by dreamsyntax** [comment]
@TellowKrinkle any chance we can get this in soon? I highly anticipate/want this change in master. If you are busy I'll open a second PR with the comment suggestions post merge.

**Review by iwubcode — APPROVED** [review]

**Pull request merged**